effekt/jvm/src/main/scala/effekt/Server.scala

```diff
@@ -231,10 +234,11 @@ trait LSPServer extends kiama.util.Server[Tree, EffektConfig, EffektError] with
     }
   } yield res
 
-  def needsUpdate(annotated: (ValueType, Effects), inferred: (ValueType, Effects))(using Context): Boolean = {
+  def needsUpdate(annotated: (List[ValueType], Effects), inferred: (List[ValueType], Effects))(using Context): Boolean = {
     val (tpe1, effs1) = annotated
     val (tpe2, effs2) = inferred
-    tpe1 != tpe2 || effs1 != effs2
+
+    tpe1.size != tpe2.size || tpe1.zip(tpe2).forall { case (t1, t2) => t1 != t2 } || effs1 != effs2
```

Comment:
please don't write huge boolean expressions like this.

```suggestion
    val differentArity = tpes1.size != tpes2.size 
    val differentEffects = effs1 != effs2 
    def differentTypes = (tpes1 zip tpes2).forall { case (tpe1, tpe2) => tpe1 != tpe2 }
    
    differentArity || differentEffects || differentTypes
```